src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from "react";
+import { fetchSingleArticle } from "../api";
+import { useParams } from "react-router-dom";
+import Button from "@mui/material/Button";
+import HourglassBottomTwoToneIcon from "@mui/icons-material/HourglassBottomTwoTone";
+
+const SingleArticle = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [singleArticle, setSingleArticle] = useState({});
+  const [commentCount, setCommentCount] = useState(0);
+  const [likesCount, setLikesCount] = useState(0);
+  const [clickLike, setClickLike] = useState(false);
+  const { article_id } = useParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetchSingleArticle(article_id).then((article) => {
+      setSingleArticle(article);
+      setCommentCount(article.comment_count);
+      setIsLoading(false);
+    });
+  }, [article_id]);
+
+  const sqlDateFormatter = (created_at) => {
+    return new Date(created_at).toLocaleString();
+  };
```

Comment:
This could maybe exist as a 'util' function and seperated out from this component?

Reply:
separated